File: Ruby/Work/name.rb
#!/usr/bin/ruby

#####################
#
# 	Created by: breton 
#	File: name.rb
#
#####################

class Name
    def initialize(first, middle, last)
        @first = first
        @middle = middle
        @last = last
        @@count += 1
    end
    
    attr_reader :first, :middle, :last
    attr_writer :first, :middle, :last
    
    def to_s
        print (@first + " " + @middle + " " + @last)
    end
end

#objects
aName = Name.new("Andres", "", "Breton")
aName.first = "Joan"    #replaces attribute
aName.middle = "Barbara"
aName.last = "Smith"
print aName.to_s, "\n"

#print from readable attributes
print aName.first
print aName.middle
print aName.last

File: Python/ObjectOP/name.py
#!/usr/bin/python

#Declare class
class Name:
    #constructor methods ("function")- instantiation => create instances in class
    def __init__(self, first, middle, last): #self parameter refers to current instance in class, always needed
        self.first = first #self.first is 1st field
        self.middle = middle #self.middle is 2nd field
        self.last = last #self.last is 3rd field


aName = Name("Mary", "Liz", "Smith") #create instance of class (object)
yourName = Name("Andres", "Breton") #another instance
